switch: cleanup and improvements

Cpasjuste opened the pull request:
- add cmake support
- add filesystem support
- add power support
- add single joycon support (start and select == L/R stick  and +/- in single joycon mode)
- update to new hid interface
- use pthread

carstene1ns commented on src/thread/switch/SDL_systhread.c:
```diff
@@ -61,24 +61,41 @@ void SDL_SYS_SetupThread(const char *name)
+        value = 0x2D;
+    }
+    else if (priority == SDL_THREAD_PRIORITY_HIGH) {
+        value = 0x2B;
```
The audio thread will run with high priority, is this still high enough (had `0x1C` before)?
Have you tested it will not lead to stutter or such things?

Cpasjuste replied:
Hi @carstene1ns , no i didn't test it yet, i'm in the process of doing so (backing up nand before in case of the new update nag). I'm going to test on each of my ports before merging.

Cpasjuste committed:
switch: add cmake support
switch: update for latest sdl2 changes
switch: update for latest libnx hid changes

Cpasjuste committed:
switch: add gitignore from sdl2 main

Cpasjuste committed:
switch: add filesystem support

Cpasjuste committed:
switch: add power support

Cpasjuste committed:
switch: use pthreads

Cpasjuste committed:
switch: update keyboard interface to new hid api

Cpasjuste committed:
switch: fix mouse interface

Cpasjuste committed:
switch: update touch interface to new hid api

Cpasjuste committed:
switch: fix wrong keyboard keys events

Cpasjuste committed:
switch: use getcwd for SDL_GetPrefPath

Cpasjuste committed:
switch: fix power interface

Cpasjuste committed:
switch: update configure.ac

Cpasjuste committed:
switch: update configure

Cpasjuste committed:
switch: cleanup joystick interface

Cpasjuste committed:
switch: handle single joycons

R-YaTian committed:
Improve single joycons mode (#16)

* Set NpadJoyHoldType_Horizontal for single joycons

* Improve single joycons mode

encounter commented:
Based on this PR, I updated to SDL 2.0.22 and made additional fixes: https://github.com/encounter/SDL/tree/switch-sdl2-2.0.22-dev

I haven't verified all functionality, just basic SDL_Renderer and audio use.

(Also I believe the `configure` changes got lost, I'm using CMake currently.)

WinterMute commented:
merged to https://github.com/devkitPro/SDL/commits/switch-sdl2
